### ISSUE TITLE
I cannot get stdin formatting to work

### ISSUE THREAD

**Issue by skorokithakis** [ambv/black#276]
Operating system: Ubuntu 17.10
Python version: 3.6.3
Black version: 18.5b1
Does also happen on master: Yes

```
02:02:52 $ black -
Usage: black [OPTIONS] [SRC]...

Error: Invalid value for "src": Path "-" is not readable.
```

Shouldn't this work? Am I being super dumb?

**Comment by ambv** [comment]
Interesting. I cannot reproduce this.
```sh
$ black -
print('hey')
print('ho')
# I'm going to CTRL+D now
print("hey")
print("ho")
# I'm going to CTRL+D now
reformatted -
```

I'm not sure what can be the case here. In the code we specifically allow dash as a special case in Click:
```py3
click.argument(
    "src",
    nargs=-1,
    type=click.Path(
        exists=True, file_okay=True, dir_okay=True, readable=True, allow_dash=True  # <-- see here
    ),
)
```

What version of Click do you have on your machine? Maybe *that* is too old?

**Comment by Grokzen** [comment]
I can confirm that the same method used by @ambv works on my end as well with basically the same setup as OP.

**Comment by skorokithakis** [comment]
I had 6.2 indeed. You may want to specify that you need a version higher than X in your setup.py.

**Comment by skorokithakis** [comment]
It works with 6.7, I will leave this ticket open as "specify `click>=6.5` in the setup.py (I tested it and it doesn't work with 6.4 or earlier).

**Comment by ambv** [comment]
@skorokithakis, instead of an issue, open a pull request! We're in this together :)

**Comment by skorokithakis** [comment]
Bah, good idea, it completely slipped my mind.

**Comment by skorokithakis** [comment]
Done in #284!

**Comment by ambv** [comment]
Fixed by #284. Thank you!
